pkgs/development/tools/comby/default.nix

```diff
@@ -0,0 +1,55 @@
+{ ocamlPackages, fetchFromGitHub, lib, zlib, pkg-config }:
+let
+  mkCombyPackage = { pname, extraBuildInputs ? [ ] }:
+    ocamlPackages.buildDunePackage rec {
+      inherit pname;
+      version = "1.5.1";
+      useDune2 = true;
+      minimumOcamlVersion = "4.08.1";
+
+      src = fetchFromGitHub {
+        owner = "comby-tools";
+        repo = "comby";
+        rev = version;
+        sha256 = "1ipfrr6n1jyyryhm9zpn8wwgzfac1zgbjdjzrm00qcwc17r8x2hf";
+      };
+
+      nativeBuildInputs = [
+        pkg-config
+        ocamlPackages.ppx_deriving
+        ocamlPackages.ppx_deriving_yojson
+        ocamlPackages.ppx_sexp_conv
+        ocamlPackages.ppx_jane
+        ocamlPackages.ppx_expect
+        ocamlPackages.dune-configurator
+      ];
+
+      buildInputs = [
```

Comment:
We should further reduce the build inputs `mkCombyPackage` passes by default — comby-kernel only uses a fraction of the (native) build inputs comby uses.

Reply:
I’m not sure how to find out which packages a package depends on. I checked the `dune` file in every subpackage, but maybe there’s a more efficient way of doing this? Anyway, I gave it a shot, could you check?

Reply:
I usually check the ones it lists in its `.opam` file.